command/cp.go

```diff
@@ -537,18 +539,20 @@ func (c Copy) doDownload(ctx context.Context, srcurl *url.URL, dsturl *url.URL)
 	if err != nil {
 		// file must be closed before deletion
 		file.Close()
-		dErr := dstClient.Delete(ctx, dsturl)
+		dErr := dstClient.Delete(ctx, &url.URL{Path: file.Name(), Type: dsturl.Type})
 		if dErr != nil {
 			printDebug(c.op, dErr, srcurl, dsturl)
 		}
 		return err
 	}
-	defer file.Close()
 
 	if c.deleteSource {
 		_ = srcClient.Delete(ctx, srcurl)
 	}
 
+	file.Close()
```

Comment:
we are closing the file for both error and success cases. can we move this block after this line?https://github.com/peak/s5cmd/blob/15fdef5620567a1ec379ce024d0cb10985d72d33/command/cp.go#L538

thus we can remove line #541

```go
       size, err := srcClient.Get(ctx, srcurl, file, c.concurrency, c.partSize)
       // close after read
       file.Close()

       if err != nil {
		// file must be closed before deletion
		dErr := dstClient.Delete(ctx, &url.URL{Path: file.Name(), Type: dsturl.Type})
		if dErr != nil {
			printDebug(c.op, dErr, srcurl, dsturl)
		}
		return err
	}
```

Reply:
Yes, we can call `file.Close()` only once. I will move it to below of that line.